anndata/_core/descriptors.py

```diff
@@ -0,0 +1,90 @@
+from typing import TYPE_CHECKING, MutableMapping, Optional
+from collections import OrderedDict
+from copy import copy
+
+from anndata.compat import _overloaded_uns, OverloadedDict, _slice_uns_sparse_matrices
+from typing_extensions import Literal
+import pandas as pd
+from anndata._core.views import DataFrameView, DictView
+
+if TYPE_CHECKING:
+    from anndata._core.anndata import AnnData
+
+
+class DataFrameDescriptor:
+    """\
+    One-dimensional annotation of observations or features.
+
+    Parameters
+    ----------
+    attr:
+        Named of the attribute.
+    """
+
+    def __init__(self, attr: Literal["obs", "var"]):
+        self._attr = attr
+
+    def __get__(self, adata: "AnnData", objtype: Optional[type] = None) -> pd.DataFrame:
+        if not adata.is_view:
+            return adata._obs if self.attr == "obs" else adata._var
+
+        adata_ref = adata._adata_ref
+        if self.attr == "obs":
+            container_full = adata_ref.obs
+            container = container_full.iloc[adata._oidx]
+        elif self.attr == "var":
+            container_full = adata_ref.var
+            container = adata_ref.var.iloc[adata._vidx]
+        else:
+            raise NotImplementedError(self.attr)
+
+        # we don't care about uns at this point
```

Comment:
Why don't we care about `uns` here?

Reply:
Ah, because you're being lazy in copying. I think in practice, many times that someone is accessing a categorical on a view they also want to get it's colors from uns because they are plotting.